crates/postgres/src/store.rs

```diff
@@ -0,0 +1,926 @@
+            });
+
+        let db = get_db().await.unwrap();
+
+        let wallet_name = get_wallet_name().unwrap();
+
+        let mut store = Store::new(db, Some(wallet_name)).await.expect("create new memory db store");
+
+
+
+        for changeset in test_changesets{
+            store.write_changes(&changeset).await.expect("write changeset");
+        }
+
+        let agg_changeset = store.load_changes().await.expect("aggregated changeset");
+
+        assert_eq!(agg_changeset, Some(agg_test_changesets));
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn insert_and_load_aggregate_changesets_with_confirmation_height_anchor() -> anyhow::Result<()>
+    {
+        let (test_changesets, agg_test_changesets) =
+            create_test_changesets(&|height, _time, hash| ConfirmationHeightAnchor {
+                confirmation_height: height,
+                anchor_block: (height, hash).into(),
+            });
+
+        let db = get_db().await.unwrap();
+
+        let wallet_name = get_wallet_name().unwrap();
+
+        let mut store = Store::new(db, Some(wallet_name)).await.expect("create new memory db store");
+
+
+
+        for changeset in test_changesets{
+            store.write_changes(&changeset).await.expect("write changeset");
+        }
+
+        let agg_changeset = store.load_changes().await.expect("aggregated changeset");
+
+        assert_eq!(agg_changeset, Some(agg_test_changesets));
+        Ok(())
+    }
+    fn get_wallet_name() -> anyhow::Result<String> {
+        let (ext_desc, int_desc) = get_desc();
+        let wallet_name = bdk_wallet::wallet::wallet_name_from_descriptor(
+            ext_desc,
+            Some(int_desc),
+            Network::Testnet,
+            &Secp256k1::new(),
+        )
+            .unwrap();
+
+        Ok(wallet_name)
+    }
+
+    async fn get_db() -> anyhow::Result<Pool<Postgres>> {
+        let db_url = "postgresql://postgres:mysecretpassword@localhost:5432/postgres";
+        let db = PgPoolOptions::new()
+            .max_connections(5)
+            .connect(&db_url)
+            .await.unwrap();
+        Ok(db)
+    }
+    #[tokio::test]
+    async fn insert_and_load_aggregate_changesets_with_blockid_anchor() -> anyhow::Result<()> {
+        let (test_changesets, agg_test_changesets) =
+            create_test_changesets(&|height, _time, hash| BlockId { height, hash });
+
+        let db = get_db().await.unwrap();
+
+        let wallet_name = get_wallet_name().unwrap();
+
+        let mut store = Store::new(db, Some(wallet_name)).await.expect("create new db store");
+
+        for changeset in test_changesets {
+            store.write_changes(&changeset).await.expect("write changeset");
+
+        }
+
+        let agg_changeset = store.load_changes().await.expect("aggregated changeset");
+
+        assert_eq!(agg_changeset, Some(agg_test_changesets));
+        Ok(())
+    }
+
+    fn get_desc() -> (&'static str,&'static str){
```

Comment:
I would love if we could move the tests helper functions to the top before the test functions